build-advisories/src/Roave/SecurityAdvisories/VersionConstraint.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+namespace Roave\SecurityAdvisories;
+
+/**
+ * A simple version constraint - naively assumes that it is only about ranges like ">=1.2.3,<4.5.6"
+ */
+final class VersionConstraint
+{
+    const CLOSED_RANGE_MATCHER    = '^>(=?)\s*((\d+.)*\d+)\s*,\s*<(=?)\s*((\d+.)*\d+)$';
+    const LEFT_OPEN_RANGE_MATCHER = '^<(=?)\s*((\d+.)*\d+)$';
+    const RIGHT_OPEN_RANGE_MATCHER = '^>(=?)\s*((\d+.)*\d+)$';
+
+    /**
+     * @var string
+     */
+    private $constraintString;
+
+    /**
+     * @var bool whether this constraint is a simple range string: complex constraints currently cannot be compared
+     */
+    private $isSimpleRangeString = false;
+
+    /**
+     * @var bool whether the lower bound is included or excluded
+     */
+    private $lowerBoundIncluded = false;
+
+    /**
+     * @var Version|null the upper bound of this constraint, null if unbound
+     */
+    private $lowerBound;
+
+    /**
+     * @var bool whether the upper bound is included or excluded
+     */
+    private $upperBoundIncluded = false;
+
+    /**
+     * @var Version|null the upper bound of this constraint, null if unbound
+     */
+    private $upperBound;
+
+    /**
+     * @param string $constraintString
+     */
+    private function __construct($constraintString)
+    {
+        $this->constraintString = (string) $constraintString;
+    }
+
+    /**
+     * @param string $versionConstraint
+     *
+     * @return self
+     *
+     * @throws \InvalidArgumentException
+     */
+    public static function fromString($versionConstraint)
+    {
+        $instance = new self($versionConstraint);
+
+        if (preg_match('/' . self::CLOSED_RANGE_MATCHER . '/', $instance->constraintString, $matches)) {
+            $instance->lowerBoundIncluded  = (bool) $matches[1];
+            $instance->upperBoundIncluded  = (bool) $matches[4];
+            $instance->lowerBound          = Version::fromString($matches[2]);
+            $instance->upperBound          = Version::fromString($matches[5]);
+            $instance->isSimpleRangeString = true;
```

Comment:
you should return the instance here instead of doing the next matches too (it cannot be both a closed range and a left open range)